kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/CheckpointV2ReadSuite.scala

```diff
@@ -0,0 +1,213 @@
+
+  def validateSnapshot(path: String, snapshotFromSpark: Snapshot): Unit = {
+    // Create a snapshot from Spark connector and from kernel.
+    val snapshot = latestSnapshot(path)
+    val snapshotImpl = snapshot.asInstanceOf[SnapshotImpl]
+
+    // Validate metadata/protocol loaded correctly from top-level v2 checkpoint file.
+    val expectedMetadataId =
+      DeltaTable.forPath(path).detail().select("id").collect().head.getString(0)
+    assert(snapshotImpl.getMetadata.getId == expectedMetadataId)
+    assert(snapshotImpl.getProtocol.getMinReaderVersion ==
+      snapshotFromSpark.protocol.minReaderVersion)
+    assert(snapshotImpl.getProtocol.getMinWriterVersion ==
+      snapshotFromSpark.protocol.minWriterVersion)
+    assert(snapshotImpl.getProtocol.getReaderFeatures.asScala.toSet ==
+      snapshotFromSpark.protocol.readerFeatureNames)
+    assert(snapshotImpl.getProtocol.getWriterFeatures.asScala.toSet ==
+      snapshotFromSpark.protocol.writerFeatureNames)
+    assert(snapshot.getVersion(defaultTableClient) == snapshotFromSpark.version)
+
+    // Validate AddFiles from sidecars found against Spark connector.
+    val scan = snapshot.getScanBuilder(defaultTableClient).build()
+    val foundFiles =
+      collectScanFileRows(scan).map(InternalScanFileUtils.getAddFileStatus).map(
+        _.getPath.split('/').last).toSet
+    val expectedFiles = snapshotFromSpark.allFiles.collect().map(_.path).toSet
+    assert(foundFiles == expectedFiles)
+  }
+
+  test("v2 checkpoint support") {
+    supportedFileFormats.foreach { format =>
+      withTempDir { path =>
+        val tbl = "tbl"
+        withTable(tbl) {
+          // Create table.
+          withSQLConf(DeltaSQLConf.CHECKPOINT_V2_TOP_LEVEL_FILE_FORMAT.key -> format,
+            "spark.databricks.delta.clusteredTable.enableClusteringTablePreview" -> "true") {
+            spark.sql(s"CREATE TABLE $tbl (a INT, b STRING) USING delta CLUSTER BY (a) " +
+              s"LOCATION '$path' " +
+              s"TBLPROPERTIES ('delta.checkpointInterval' = '2', 'delta.checkpointPolicy'='v2')")
+            spark.sql(s"INSERT INTO $tbl VALUES (1, 'a'), (2, 'b')")
+            spark.sql(s"INSERT INTO $tbl VALUES (3, 'c'), (4, 'd')")
+            spark.sql(s"INSERT INTO $tbl VALUES (5, 'e'), (6, 'f')")
+          }
+
+          // Validate snapshot and data.
+          validateSnapshot(path.toString, DeltaLog.forTable(spark, path.toString).update())
+          checkTable(
+            path = path.toString,
+            expectedAnswer = (1 to 6).map(i => TestRow(i, (i - 1 + 'a').toChar.toString))
+          )
+
+          // Remove some files from the table, then add a new one.
+          spark.sql(s"DELETE FROM $tbl WHERE a=1 OR a=2")
+          spark.sql(s"INSERT INTO $tbl VALUES (7, 'g'), (8, 'h')")
+
+          // Validate snapshot and data.
+          validateSnapshot(path.toString, DeltaLog.forTable(spark, path.toString).update())
+          checkTable(
```

Comment:
This should also be covered in `SnapshotManagerSuite` - this suite is end-to-end just to cover the overall read process.

Reply:
Yes its already covered in UTs. It will be good to assert that the tests are exactly testing the scenario that they were supposed to test. Helps with avoiding and configuration related issues e.g. [this one](https://github.com/delta-io/delta/pull/2826#discussion_r1578626599).

Reply:
Got it, this is now in the test as well.